### PR TITLE
Update libreconv to include timeout

### PR DESCRIPTION
**Pull request by migueltaoliveira**
SL-5235

### CODE REVIEW

File: lib/libreconv.rb [hunk 1]
```diff
@@ -63,19 +67,23 @@ def convert
...
-      output, error, status =
-        if RUBY_PLATFORM =~ /java/
-          Open3.capture3(*command)
-        else
-          Open3.capture3(command_env, *command, unsetenv_others: true)
-        end
+      cmd = if RUBY_PLATFORM =~ /java/
+        Mixlib::ShellOut.new(*command, :timeout => @timeout_secs)
+      else
+        Mixlib::ShellOut.new(*command, :environment => command_env, :timeout => @timeout_secs)
```
Comment (dmbento) [inline lib/libreconv.rb]:
```suggestion
        Mixlib::ShellOut.new(*command, timeout: @timeout_secs)
      else
        Mixlib::ShellOut.new(*command, environment: command_env, timeout: @timeout_secs)
```

File: lib/libreconv.rb [hunk 2]
```diff
@@ -10,17 +10,19 @@
 # Convert office documents using LibreOffice / OpenOffice to one of their supported formats.
 module Libreconv
   class ConversionFailedError < StandardError; end
+  class ConversionTimeoutError < StandardError; end
```
Comment (dmbento) [inline lib/libreconv.rb:13]: should this extend from the failed?

Comment (migueltaoliveira) [reply lib/libreconv.rb:13]: It's better to create another.

Comment (dmbento) [reply lib/libreconv.rb:13]: semantically, the conversion failed, that's why I'm asking. might be confusing otherwise

Comment (migueltaoliveira) [reply lib/libreconv.rb:13]: On other side, timeout it's not retriable, that's why I would prefer to keep it as a different error!

**Review by dmbento — APPROVED** [review]

**Pull request merged**